extensions/wikia/Wall/WallHooksHelper.class.php

```diff
@@ -529,86 +529,51 @@ static public function onSkinSubPageSubtitleAfterTitle( $title, &$ptext, &$cssCl
 	/**
 	 * @brief Adds an action button on user talk archive page
 	 *
-	 * @param $response
-	 * @param $ns
-	 * @param $skin
-	 * @return bool
+	 * @param array $button 'Edit' button and dropdown
+	 * @return bool true
 	 *
 	 * @author Andrzej 'nAndy' Łukaszewski
 	 */
-	static public function onPageHeaderIndexAfterActionButtonPrepared( $response, $ns, $skin ) {
+	static public function onPageHeaderIndexAfterActionButtonPrepared( array &$button ) {
 		$app = F::App();
 		$helper = new WallHelper();
 
 		if ( !empty( $app->wg->EnableWallExt ) ) {
 			$title = $app->wg->Title;
+			$ns = $title->getNamespace();
 			$parts = explode( '/', $title->getText() );
-			$action = $response->getVal( 'action' );
-			$dropdown = $response->getVal( 'dropdown' );
 			$canEdit = $app->wg->User->isAllowed( 'editwallarchivedpages' );
 
-			if ( $ns === NS_USER_WALL
+			if (
+				$ns === NS_USER_WALL
 				&& $title->isSubpage()
 				&& !empty( $parts[1] )
 				&& mb_strtolower( str_replace( ' ', '_', $parts[1] ) ) === mb_strtolower( $helper->getArchiveSubPageText() )
 			) {
-				// user talk archive
```

Comment:
With regards to the first difference the namespace is always derived from `$wgTitle`, see the value of `$ns` being set [here](https://github.com/Wikia/app/blob/42a351cba9082b2eff9a69ee8b5c92df077d1203/skins/oasis/modules/PageHeaderController.class.php#L181-L201). It's essentially the same condition for both, so either both or neither conditional is executed no matter what.

Are there subpages in NS_WALL other than user talk archives that this should handle or used to handle? That's the only reason I can think of for the two conditions to exist in the first place.